src/front/glsl/lex.rs

```diff
@@ -16,278 +16,272 @@ fn consume_any(input: &str, what: impl Fn(char) -> bool) -> (&str, &str, usize)
+                match word {
+                    "layout" => Some(Token::Layout(meta)),
+                    "in" => Some(Token::In(meta)),
+                    "out" => Some(Token::Out(meta)),
+                    "uniform" => Some(Token::Uniform(meta)),
+                    "flat" => Some(Token::Interpolation((meta, crate::Interpolation::Flat))),
+                    "noperspective" => {
+                        Some(Token::Interpolation((meta, crate::Interpolation::Linear)))
+                    }
+                    "smooth" => Some(Token::Interpolation((
                         meta,
                         crate::Interpolation::Perspective,
                     ))),
-                    rest,
-                ),
-                "centroid" => (
-                    Some(Token::Interpolation((meta, crate::Interpolation::Centroid))),
-                    rest,
-                ),
-                "sample" => (
-                    Some(Token::Interpolation((meta, crate::Interpolation::Sample))),
-                    rest,
-                ),
-                // values
-                "true" => (Some(Token::BoolConstant((meta, true))), rest),
-                "false" => (Some(Token::BoolConstant((meta, false))), rest),
-                // jump statements
-                "continue" => (Some(Token::Continue(meta)), rest),
-                "break" => (Some(Token::Break(meta)), rest),
-                "return" => (Some(Token::Return(meta)), rest),
-                "discard" => (Some(Token::Discard(meta)), rest),
-                // types
-                "void" => (Some(Token::Void(meta)), rest),
-                word => {
-                    let token = match parse_type(word) {
-                        Some(t) => Token::TypeName((meta, t)),
-                        None => Token::Identifier((meta, String::from(word))),
-                    };
-                    (Some(token), rest)
+                    "centroid" => {
+                        Some(Token::Interpolation((meta, crate::Interpolation::Centroid)))
+                    }
+                    "sample" => Some(Token::Interpolation((meta, crate::Interpolation::Sample))),
+                    // values
+                    "true" => Some(Token::BoolConstant((meta, true))),
+                    "false" => Some(Token::BoolConstant((meta, false))),
+                    // jump statements
+                    "continue" => Some(Token::Continue(meta)),
+                    "break" => Some(Token::Break(meta)),
+                    "return" => Some(Token::Return(meta)),
+                    "discard" => Some(Token::Discard(meta)),
+                    // types
+                    "void" => Some(Token::Void(meta)),
+                    word => {
+                        let token = match parse_type(word) {
+                            Some(t) => Token::TypeName((meta, t)),
+                            None => Token::Identifier((meta, String::from(word))),
```

Comment:
This is concerning for performance. We don't want to be heap-allocating `String::from` on every token, that's really bad.
Notably, it's been there before the PR, so technically unrelated.
The token itself should just borrow strings. See how WGSL frontend does it.

Do you want to try making this happen in this PR? Or follow up?

Reply:
i think this relates a bit to how pomelo handles tokens, so better in separate PR